services/device-registry-base/src/main/java/org/eclipse/hono/service/management/device/Status.java

```diff
@@ -0,0 +1,75 @@
+/*******************************************************************************
+ * Copyright (c) 2019 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+
+package org.eclipse.hono.service.management.device;
+
+import java.time.Instant;
+
+import org.eclipse.hono.annotation.HonoTimestamp;
+import org.eclipse.hono.util.RegistryManagementConstants;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/**
+ * A status used in Device value object.
+ */
+public final class Status {
+
+    @JsonProperty(RegistryManagementConstants.FIELD_STATUS_CREATION_DATE)
+    @JsonInclude(value = JsonInclude.Include.NON_EMPTY)
```

Comment:
I guess this should be `JsonInclude.Include.NOT_NULL`. AFAIK `JsonInclude.Include.NON_EMPTY` is used for collections only, isn't it?

Reply:
Non empty is not limited to collections, it's simply that when serializing the object, if the value is empty, the field will be omitted. With `NOT_NULL` the field won't be omitted as long as it's not null.